internal/vehicle/niu.go

```diff
@@ -60,7 +60,7 @@ func NewNiuFromConfig(other map[string]interface{}) (api.Vehicle, error) {
 		serial:   strings.ToUpper(cc.Serial),
 	}
 
-	v.chargeStateG = provider.NewCached(v.chargeState, cc.Cache).FloatGetter()
+	v.API = niu.New(provider.NewCached(v.batteryAPI, cc.Cache).InterfaceGetter())
```

Comment:
Das ist jetzt ein bisschen sehr wild. Du machst hier das caching um es in der anderen Datei zu verwenden? Bitte nicht!

Reply:
Nur für mich zum Verständnis: :-)
Meinst Du damit, dass ich die vehicle API Funktionen aus der niu/api.go wieder in die niu.go holen soll?  
Die niu/api.go dann löschen?

Reply:
Oder einfach:
	v.API = niu.New(v.batteryAPI)

?

Reply:
Replaced by v.API = niu.New(v.batteryAPI)

Reply:
Nein- ich möchte nicht, dass eine Methode aus dem Unterordner von übergebenem Verhalten aus dem Oberordner abhängt. Entweder: machst Du alles im Unterordner (siehe VW) oder Du machst das Caching im Oberordner, hast dafür im Unterordner dafür aber zwei saubere Funktionen. Die werden im Oberordner dann eben beide gecached. Nicht super, aber auch möglich. 
Sonst im Unterordner in api und provider trennen. Eines implementiert das Fahrzeugseitige API, das Andere macht es für evcc nutzbar.

Oder ich merge und räume es auf ;)